### ISSUE TITLE
Support named + anonymous AMD

### ISSUE THREAD

**Issue by guybedford** [systemjs/systemjs#1116]
So that a module of the form:

```javascript
define('x', ...);

define(function() {
  return 'asdf';
});
```

Will both define into the registry as `System.set(System.decanonicalize('x'), ...` and as the current module value of `asdf`.

**Comment by guybedford** [comment]
This named plus anonymous feature could also potentially be extended to System.register support too.

**Comment by guybedford** [comment]
Implemented pending build support.

**Comment by guybedford** [comment]
This will not be done for System.register as it is a terrible idea.

**Comment by guybedford** [comment]
Released in 0.19.23.
